hles/transformation/src/main/scala/org/broadinstitute/monster/dap/TsvUtils.scala

```diff
@@ -0,0 +1,70 @@
+package org.broadinstitute.monster.dap
+
+import java.io.StringWriter
+
+import com.spotify.scio.values.SCollection
+
+import io.circe._
+import io.circe.parser._
+
+import com.opencsv.CSVWriter
+
+object TsvUtils {
+
+  def jsonToTsv(
+    messages: SCollection[String],
+    headers: Seq[String],
+    jsonToTsvRowTransform: JsonObject => JsonObject,
+    description: String,
+    outputPrefix: String
+  ) =
+    messages
+      .transform(s"Convert '$description' JSON to TSV rows")(
+        _.map(row => jsonStringToTsvString(row, headers, jsonToTsvRowTransform))
+      )
+      .withName(s"Write '$description' TSV messages to '$outputPrefix'")
+      .saveAsTextFile(
+        outputPrefix,
+        suffix = ".tsv",
+        numShards = 0,
+        header = Some(headers.mkString("\t"))
+      )
+
+  def stringSeqToTsvRow(cells: Iterable[String]): String = {
+    val rowBuilder: StringWriter = new StringWriter()
+    val writer = new CSVWriter(rowBuilder, '\t', '"', '"', "\n")
+    // the second argument tells the writer how to handle quoting.
+    // if it's false, it will only wrap cells containing quote/newline/separator characters in quotes.
+    writer.writeNext(cells.toArray, false)
+
+    rowBuilder.toString
+  }
+
+  // presumes that the element is not itself a json object (i.e. json objects we map this function across must be flat)
+  def jsonElementToString(element: Json): String = {
```

Comment:
⛏️ Would it make sense to add a `case _` to catch and throw in circumstances where this function is passed an object?